src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -750,10 +807,25 @@ public DependencyGraphResolver(RestoreCollectorLogger logger, RestoreRequest res
 
                             var newGraphNode = new GraphNode<RemoteResolveResult>(actualDep.LibraryRange);
                             newGraphNode.Item = findLibraryEntryCache[chosenItemRangeIndex].Item;
-                            currentGraphNode.InnerNodes.Add(newGraphNode);
-                            newGraphNode.OuterNode = currentGraphNode;
 
-                            if (newGraphNode.Item.Key.Type != LibraryType.Project && newGraphNode.Item.Key.Type != LibraryType.ExternalProject && newGraphNode.Item.Key.Type != LibraryType.Unresolved && !versionConflicts.ContainsKey(chosenItemRangeIndex) && dep.SuppressParent != LibraryIncludeFlags.All && dep.LibraryRange.VersionRange != null && !dep.LibraryRange.VersionRange!.Satisfies(newGraphNode.Item.Key.Version))
+                            if (chosenItem.IsCentrallyPinnedTransitivePackage)
+                            {
+                                newGraphNode.Disposition = Disposition.Accepted;
+                                newGraphNode.Item.IsCentralTransitive = true;
+                                newGraphNode.OuterNode = rootGraphNode;
+                                rootGraphNode.InnerNodes.Add(newGraphNode);
+                            }
+                            else
+                            {
+                                newGraphNode.OuterNode = currentGraphNode;
+                                currentGraphNode.InnerNodes.Add(newGraphNode);
+                            }
+                            if (isCentralPackageTransitivePinningEnabled && !downgrades.ContainsKey(chosenItemRangeIndex) && !RemoteDependencyWalker.IsGreaterThanOrEqualTo(chosenItem.LibraryDependency.LibraryRange.VersionRange, dep.LibraryRange.VersionRange))
+                            {
+                                downgrades.Add(chosenItem.LibraryRangeIndex, (currentGraphNode, dep));
+                            }
+
+                            if (newGraphNode.Item.Key.Type != LibraryType.Project && newGraphNode.Item.Key.Type != LibraryType.ExternalProject && newGraphNode.Item.Key.Type != LibraryType.Unresolved && !versionConflicts.ContainsKey(chosenItemRangeIndex) && dep.SuppressParent != LibraryIncludeFlags.All && dep.LibraryRange.VersionRange != null && !dep.LibraryRange.VersionRange!.Satisfies(newGraphNode.Item.Key.Version) && !downgrades.ContainsKey(chosenItemRangeIndex))
```

Comment:
I have a really really tough time processing this. 

Comments would be helpful for conditions that include this many and statements.